build.gradle

```diff
@@ -10,6 +10,13 @@ plugins {
     id "io.morethan.jmhreport" apply false
 }
 
+if (!JavaVersion.current().isJava11Compatible()) {
+    throw new GradleException("Java 11 or higher is required to build. " +
+            "Please download it from https://adoptopenjdk.net/. If you believe you already " +
```

Comment:
I am against this endorsement. Albeit all the great work that AdoptOpenJDK did, I cannot recommend their distribution. At least as they are not [TCK-compliant](https://adoptopenjdk.net/quality.html#jck).

Reply:
I mainly picked it because Gradle defaults to downloading it, and GitHub Actions has a plan to as well https://github.com/actions/setup-java/issues/98. I don't know why, but I'm personally OK with any URL here as long as it's stable, which official OpenJDK ones aren't.

Reply:
My go-to java distribution is [Liberica](https://bell-sw.com). Another option is [Zulu](https://www.azul.com/downloads/zulu-community/) (which is default for setup-java as of now)

Reply:
I would prefer Adopt, but what about not including any concrete suggestion? Since this seems to be a point of conflict.

Reply:
I checked openjdk.java.net again for Java 11 but it seems to be as hidden as ever. So the most standard option doesn't seem great.

I have a personal preference for zulu, mainly because of the TCK thing but see the ecosystem consolidating on adopt (GitHub planning to switch from zulu to adopt does mean something), which is why I went with it here. Admittedly the TCK issue seems theoretical given they are in the end effectively the same codebase, and both have wide adoption.

I think a link is far more user friendly than no link. Given our Gradle build already automatically downloads adopt if Java 11 or 8 isn't installed, I am leaning strongly for not making a change to this url. I also wouldn't revert the change to using Gradle's toolchain feature to automatically download adopt because automatically downloading a missing JDK is way more user friendly than not.

What do you think?

Reply:
Also somewhat random but if someone can help with https://github.com/diffplug/spotless/issues/724 than any Java would work with the build. I want to do it someday (cloned spotless repo :) but probably still will take some time. After that we can remove this message - if anyone has strong issues with the message I encourage them to help with the issue since this is OSS :D

Reply:
How about mentioning several options, so it doesn't look like we're endorsing a single JDK distro?

Reply:
I think it's ok to include a link to AdoptOpenJDK. It's great to have a vendor neutral option. And this is only a recommendation for building the project (not for production usage where TCK compliance would be more of an influence).

To make clear that we don't specifically require AdoptOpenJDK:

`Please download it from https://adoptopenjdk.net/.`

==>

`One option is to download it from https://adoptopenjdk.net/.`

Reply:
thought I had seen news on the AdoptOpenJDK TCK front, found it:

https://blog.adoptopenjdk.net/2020/06/adoptopenjdk-to-join-the-eclipse-foundation/

https://projects.eclipse.org/projects/adoptium/governance:

> Scope: 
Eclipse Adoptium provides runtime binaries that are high performance, enterprise-caliber, cross-platform, open-source licensed, Java SE TCK-tested and compliant for general use across the Java ecosystem

Reply:
the real facepalm is the name "Adoptium"